### ISSUE TITLE
Namespace or template inside ifdef causing false syntax errors

### ISSUE THREAD

**Issue by mayshukla** [ualberta-smr/varclang#35]
Example:
```cpp
#ifdef A
namespace foo {
  int i;
}
#endif
```

Output:

```
tmp.cpp:2:1: error: expected unqualified-id
namespace foo {
^
namespace foo {
    int i; // In context: A
}// In context: A
1 error generated.
```

Note that this error is only emitted if `namespace` is the first thing inside the ifdef

**Comment by mayshukla** [comment]
This issue also occurs when there is a `template` keyword immediately after `#ifdef`

For example:
```cpp
#ifdef A
template <typename T>
void foo() {}
#endif
```

Output:
```
error: expected unqualified-id
template <typename T>
^
1 error generated.
```

**Comment by mayshukla** [comment]
Fixed in 3c8ab40e9014a1b3fe35740f0a7ed426dd3ea059
